CONTRIBUTING.md

```diff
@@ -151,6 +151,36 @@ Running the script for a project in `sdk\tables` would look like this:
 eng\scripts\Export-API.ps1 tables
 ```
 
+## Updating Sample Snippets
+If the specific client library has sample snippets in markdown format, they were most likely created with help of the `eng\scripts\Update-Snippets.ps1` script.
+Any changes made to the snippet markdown should be done via updating the corresponding C# snippet code and subsequently running the script.
+
+Running the script for a project, for example in `sdk\keyvault`, would look like this: 
+```
+eng\scripts\Update-Snippets.ps1 keyvault
+```
+
+When run, the code regions in the format below (where `<snippetName>` is the name of the snippet):
+```c#
+#region Snippet:<snippetName>
+//some sample code
+string snippet = "some snippet code";
+
+// Lines prefixed with the below comment format will be ignored by the snippet updater
+/*@@*/ string ignored = "this code will not appear in the snippet markdown";
```

Comment:
There's also a markdown only pattern where it will show up as regular code in markdown but be commented in C#:

```C#
//@@ snippet = "value that would never pass a test but looks good in a sample!";
```

https://github.com/Azure/azure-sdk-for-net/blob/master/eng/SnippetGenerator/DirectoryProcessor.cs#L23

They should not be used heavily because this code won't be compiled, tested, refactored, etc.